### PR TITLE
Fix rocAL Tensor build issues PR 4 - Pipeline and Pybind

### PR DESCRIPTION
**Pull request by fiona-gladwin**
- Re introduced the copy tensor API in master graph as it gives performance in tensor trainings
- Re-introduce some external API to get width and height from master graph as suggested
- Introduce API and changes required to get original ROI from tensor (Similar to TOT changes)
- Fix build issues with pybind

### CODE REVIEW

File: rocAL/rocAL/include/pipeline/master_graph.h [hunk 1]
```diff
@@ -50,6 +50,25 @@ THE SOFTWARE.
 #define MAX_NUM_ANCHORS 8732  // Num of bbox achors used in SSD training
 #define MAX_MASK_BUFFER 10000
 
+#if ENABLE_SIMD
```
Comment (rrawther) [inline rocAL/rocAL/include/pipeline/master_graph.h:53]: this seems to be missing in tot MIVIsionX. Why is this needed here

File: rocAL/rocAL/source/pipeline/master_graph.cpp [hunk 2]
```diff
@@ -460,6 +487,447 @@ MasterGraph::timing()
...
+    
+    if(_output_tensor_list[0]->info().data_type() != RocalTensorDataType::UINT8)
+        THROW("The output tensor is not of UINT8 type")
+
+    if (_output_tensor_list[0]->info().color_format() == RocalColorFormat::RGB_PLANAR)
+        return MasterGraph::copy_out_tensor_planar(out_ptr, format,multiplier0, multiplier1, multiplier2, offset0, offset1, offset2, reverse_channels, output_data_type);
+
+    _convert_time.start();
+    // Copies to the output context given by the user
+    auto dims = _output_tensor_list[0]->info().dims();
```
Comment (rrawther) [inline rocAL/rocAL/source/pipeline/master_graph.cpp]: please use another pointer to point to output_tensor_list[0]->info() and use that everywhere.

File: rocAL/rocAL/source/pipeline/master_graph.cpp [hunk 3]
```diff
@@ -460,6 +487,447 @@ MasterGraph::timing()
...
+    unsigned int n = dims[0];
+    const size_t c = dims[3];
+    const size_t h = dims[1];
+    const size_t w = dims[2];
+    const size_t single_output_image_size = _output_tensor_list[0]->info().data_size();
```
Comment (rrawther) [inline rocAL/rocAL/source/pipeline/master_graph.cpp]: rename to tensor specific name

File: rocAL/rocAL/include/pipeline/master_graph.h (hunk 1 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/include/pipeline/master_graph.h:53]: This portion was a part of [master_graph.cpp](https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/8cbd2e86fdff77faf1ce93cb4e7121010142573a/rocAL/rocAL/source/pipeline/master_graph.cpp#L41). 
We have moved the masks as constants into the master_graph.h hence we have moved the include part also from the cpp file into master_graph.h file

File: rocAL/rocAL/source/pipeline/tensor.cpp [hunk 4]
```diff
@@ -112,6 +112,8 @@ void TensorInfo::reset_tensor_roi_buffers() {
         size_t roi_size = (_layout == RocalTensorlayout::NFCHW || _layout == RocalTensorlayout::NFHWC) ? _dims[0] * _dims[1] : _batch_size; // For Sequences pre allocating the ROI to N * F to replicate in OpenVX extensions
         allocate_host_or_pinned_mem((void **)&_roi_buf, roi_size * 4 * sizeof(unsigned), _mem_type);
     }
+    if(!_original_roi_height->size()) _original_roi_height = std::make_shared<std::vector<uint32_t>>(_batch_size);
```
Comment (rrawther) [inline rocAL/rocAL/source/pipeline/tensor.cpp]: Why is changed from struct roi to width and height

File: rocAL/rocAL_pybind/rocal_pybind.cpp [hunk 5]
```diff
@@ -377,33 +377,31 @@ namespace rocal{
...
                     py::array numpy_array;
-                    switch (output_tensor.info().data_type())
+                    switch (output_tensor.data_type())
                     {
-                    case RocalTensorDataType::UINT8:
-
+                    case RocalTensorOutputType::ROCAL_UINT8:
                         numpy_array = py::array(py::buffer_info(
-                            ((unsigned char *)(output_tensor.buffer())) + idx * (info.strides()[0]/sizeof(float)),
+                            ((unsigned char *)(output_tensor.buffer())) + idx * (output_tensor.strides()[0]/sizeof(float)),
```
Comment (rrawther) [inline rocAL/rocAL_pybind/rocal_pybind.cpp]: why is this dividing by sizeof(float) for UINT8

File: rocAL/rocAL_pybind/rocal_pybind.cpp [hunk 6]
```diff
@@ -430,27 +428,27 @@ namespace rocal{
                 "at",
```
Comment (rrawther) [inline rocAL/rocAL_pybind/rocal_pybind.cpp:427]: why there are 2 separate at functions?

**Review by rrawther — CHANGES_REQUESTED** [review]
please address review comments

File: rocAL/rocAL_pybind/rocal_pybind.cpp (hunk 6 above)
Comment (fiona-gladwin) [reply rocAL/rocAL_pybind/rocal_pybind.cpp:427]: There are 2 at definitions 1 for rocalTensor and another for rocalTensorList
at in rocalTensor - Used to get an image/audio/sequence at the specified position from a batch.
at in rocalTensorList - To get a tensor at the specified position from the list.

File: rocAL/rocAL/source/pipeline/master_graph.cpp (hunk 2 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/source/pipeline/master_graph.cpp]: Done

File: rocAL/rocAL/source/pipeline/master_graph.cpp (hunk 3 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/source/pipeline/master_graph.cpp]: Done

File: rocAL/rocAL/source/pipeline/tensor.cpp (hunk 4 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/source/pipeline/tensor.cpp]: Removed these _original_roi_width and _original_roi_height as they are not used anymore

File: rocAL/rocAL_pybind/rocal_pybind.cpp (hunk 5 above)
Comment (fiona-gladwin) [reply rocAL/rocAL_pybind/rocal_pybind.cpp]: Fixed

**Comment by fiona-gladwin** [comment]
> please address review comments

@rrawther and @LakshmiKumar23 I have addressed all review comments. Please check.

**Review by rrawther — APPROVED** [review]
